restli-server/src/main/java/com/linkedin/restli/internal/server/model/ResourceModelEncoder.java

```diff
@@ -314,6 +338,53 @@ public ResourceSchema loadOrBuildResourceSchema(final ResourceModel resourceMode
     }
   }
 
+  private InputStream getResourceStreamBySearchingRestSpec(String resourceName, ClassLoader... classLoaders)
+  {
+    final String REST_SPEC_JSON_SUFFIX = "restspec.json";
+    if (!resourceName.endsWith(REST_SPEC_JSON_SUFFIX))
+    {
+      return null;
+    }
+    if (_restSpecToClassLoaderMap == null)
+    {
+      _restSpecToClassLoaderMap = new HashMap<>();
+      for(ClassLoader classLoader: classLoaders)
+      {
+        URL[] urls = ((URLClassLoader) classLoader).getURLs();
```

Comment:
Are all the classLoaders always `URLClassLoader` in this case?

Reply:
Yes. In fact it should be either AppClassLoader or ExtClassLoader which all extends URLClassLoader